### ISSUE TITLE
Map caching conflicts may cause maps to be discarded without ever being sold.

### ISSUE THREAD

**Issue by Estecka** [Estecka/mc-Shifting-Wares#11]
When the gamerules "Depleted Reroll" and "Reroll sold maps" are both enabled, it is possible for a cartographer to end up simultaneously selling two different maps of the same type :
1. A map is bought from the cartographer, without depleting the trade.
2. An adjacent trade of the same level is depleted.
3. Upon rerolling depleted trades, the second trade has a chance to yield a map of the same type as the first trade. This map willl be regenerated and have a different ID from the first one.

The villager's cache can only accomodate for one map of each type, so the two maps will compete to be registered, and intermitently overwrite each other. The older map will cause the new one to be incorrectly marked as already sold.

If the gamerule "Daily Rerolls" is disabled, this isn't expect to cause any issue, beyond the warnings and error printed into the log.

If the gamerule "Daily Rerolls" is enabled, then the next time it occurs, the newer map will be discarded, regardless of whether it has actually been sold.

In short, this will only cause issues if all three gamerules are enabled.

**Comment by Estecka** [comment]
Fixed in v3
